api/src/views/context.rs

```diff
@@ -79,13 +91,53 @@ impl UserContext {
                 user: None,
             }),
             // User is logged in and initialized, return the full user data
-            Some((Some(user_id), Some(username))) => Some(Self {
-                user_provider_id,
-                user: Some(AuthorizedUser {
-                    id: user_id,
-                    username,
-                }),
-            }),
+            Some((Some(user_id), Some(username))) => {
+                // Fetch all of the user's roles and permissions
+                let permissions_query_result = users::table
+                    // yo dawg i heard you like joins
+                    .inner_join(
+                        user_roles::table.left_join(
+                            // LEFT JOIN here so that even if the role has no
+                            // permissions, we still get its is_admin value
+                            roles::table.left_join(
+                                role_permissions::table
+                                    .left_join(permissions::table),
+                            ),
+                        ),
+                    )
+                    .select((
+                        permissions::columns::name.nullable(),
+                        roles::columns::is_admin,
+                    ))
+                    .distinct()
+                    .filter(users::columns::id.eq(user_id))
+                    .get_results::<(Option<PermissionType>, bool)>(conn)?;
```

Comment:
Im not sure if it would be easy in disel but doing an array agg on the permissions might let use remove the code block after this that makes the hash set. If its annoying im cool with leaving it as is